### ISSUE TITLE
Feature Request: MatrixChannel

### ISSUE THREAD

**Issue by dabacon** [quantumlib/Cirq#4120]
**Is your feature request related to a use case or problem? Please describe.**

`cirq.MatrixGate` is a convenient way to create a unitary gate for an arbitrary matrix.  It would be nice to have a quick way to create a channel with Krauss operators directly.  

**Describe the solution you'd like**

A new gate `cirq.MaxtrixChannel` that allows one to pass in the Krauss operators.  

```python
kill_flip = cirq.MatrixChannel([[[0, 1], [0, 0]], [[0, 0],[1, 0]]])
circuit.append(kill_flip(q0))
```

Alternatively one could reuse `cirq.MatrixGate`.

**[optional] Describe alternatives/workarounds you've considered**

Can create your own custom channel and implement `_channel_`. 

**[optional] Additional context (e.g. screenshots)**

P3 - I'm not really blocked by it, it is an idea I'd like to discuss / suggestion based on principle

**Comment by 95-martin-orion** [comment]
The [cirq-custom-channel](https://tinyurl.com/cirq-custom-channel) RFC may be of interest here. The measurement features it includes are motivated by #3241.

**Comment by dstrain115** [comment]
This is solved by custom channel.
